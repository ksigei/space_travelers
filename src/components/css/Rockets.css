.box {
  width: 100%;
  display: flex;
  margin-top: 30px;
}

.box .left {
  flex: 1;
  margin-right: 20px;
}

.box .left .rocket-img {
  width: 100%;
  height: auto;
}

.box .left .rocket-img img {
  width: 100%;
  height: 100%;
}

.box .right {
  flex: 3;
}
