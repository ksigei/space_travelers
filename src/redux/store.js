import { composeWithDevTools } from 'redux-devtools-extension';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rocketReducers from './rockets/reducers/rocketReducers';

const rootReducer = combineReducers({
  rocketReducers,
});

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk)),
);
export default store;
