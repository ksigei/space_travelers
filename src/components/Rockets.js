import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import getRocketsAction from '../redux/rockets/actions';
import './css/Rockets.css';

const Rockets = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getRocketsAction());
  }, []);
  const rockets = useSelector((state) => state.rocketReducers);
  return rockets.map((rocket) => (
    <main className="box" key={rocket.rocketId}>
      <div className="left">
        <div className="rocket-img">
          <img src={rocket.rocketImgUrl} alt="rocket" />
        </div>
      </div>
      <div className="right">
        <h4>{rocket.rocketName}</h4>
        <p>{rocket.rocketDescription}</p>
        <button
          type="button"
          className="btn btn-primary btn-md"
        >
          Reserve Rocket

        </button>
      </div>
    </main>
  ));
};

export default Rockets;
