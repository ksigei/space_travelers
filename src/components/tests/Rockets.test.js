import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../Rockets';
import store from '../../redux/store';

test('check if rockets matches the snapshot', () => {
  const tree = renderer.create(<Provider store={store}><Rockets /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
