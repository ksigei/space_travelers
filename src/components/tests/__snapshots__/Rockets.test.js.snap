// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if rockets matches the snapshot 1`] = `null`;
